docs/src/installation/Feature-flags.md

```diff
@@ -0,0 +1,62 @@
+# Feature Flags
+
+`Spotifyd` is split into a base package plus additional features that can be toggled on or off during compilation. Those can be split into two groups: The audio backend features that are responsible for playing back the music and additional functionality features, which enhance your experience using `spotifyd`.
+
+To enable an additional audio backend, pass `<audio_backend_name>_backend` as a feature flag. We currently support `alsa`, `pulseaudio` and `portaudio`.
+
+`Spotifyd` provides the following additional functionality:
+
+| Feature Flag | Description                                                                         |
+|--------------|-------------------------------------------------------------------------------------|
+| dbus_keyring | Provides password authentication over the system's keyring (supports all platforms) |
+| dbus_mpris   | Provides multimedia key support (Linux only)                                      |
+
+> __Note:__ Compiling Spotifyd with all features and the pulseaudio backend on Ubuntu would result in the following command: `cargo build --release --no-default-features --features pulseaudio_backend,dbus_keyring,dbus_mpris`
+
+## Media controls
+
+Spotifyd implements the [MPRIS D-Bus Interface Specification][mpris-specification], meaning that it can be controlled by generic media playback controllers such as [playerctl][playerctl-homepage] as well as some tools specifically designed for use with the official Spotify client such as [sp][sp-homepage].
+
+> __Note:__ Make sure to rename the service name within the `sp` script to `spotifyd`!
+
+Although the code greatly improved, this feature is still considered experimental. Make sure to open an issue if you encounter any issues while using other players to control `spotifyd`.
+
+## Audio Backends
+
+By default, the audio backend is ALSA, as ALSA is available by default on a lot of machines and usually doesn't require extra dependencies. There is also support for `pulseaudio` and `portaudio`.
+
+> __Note:__ To disable this audio backend, pass `--no-default-features` down during compilation.
+
+### PulseAudio
+
+To use PulseAudio, compile with the `--features` flag to enable
+it:
+
+```bash
+cargo build --release --features "pulseaudio_backend"
+```
+
+You will need the development package for PulseAudio, as well
+as `build-essential` or the equivalent package of your distribution.
+
+### PortAudio
+
+To use PortAudio (works on macOS), compile with the `--features` flag to enable it:
```

Comment:
Works on linux too. I'm using portaudio and (compiled with the right rust version) It works! Mabe we should remove the 'works on MacOS' part?